### PR TITLE
Add trained competence to course.

### LINKED ISSUE #670: [FEATURE] add a trained competences section in the certificate
maybe we could add a trained competences in the certificate

```
Competences
* Plugin development
* Processing algorithm implementation
* PyQGIS Yoda```

### PR DESCRIPTION
**Pull request by sonlinux**
fix #670 

- [x] Show trained competence on course detail.

competences

- [x] Show course language and trained competence on verify certificate page.

cerify_cert

- [x] Show trained competence on PDF certificate

competences

- [x] course admin screenie
course_admin

### CODE REVIEW

File: django_project/certification/models/course.py [hunk 1]
```diff
@@ -38,7 +38,15 @@ class Course(models.Model):
...
         null=True,
-        blank=True,
+        blank=True
+    )
+
+    trained_competence = models.CharField(
+        help_text=_(
+            'Trained competence e.g. Plugin development.'),
+            max_length=200,
+            null=True,
```
Comment (ismailsunni) [inline django_project/certification/models/course.py]: __[Flake8]__

[E131] continuation line unaligned for hanging indent

File: django_project/certification/models/course.py [hunk 2]
```diff
@@ -38,7 +38,15 @@ class Course(models.Model):
...
+            blank=True
```
Comment (ismailsunni) [inline django_project/certification/models/course.py]: __[Flake8]__

[E131] continuation line unaligned for hanging indent

File: django_project/certification/models/course.py [hunk 3]
```diff
@@ -38,7 +38,15 @@ class Course(models.Model):
...
         default=_('English'),
         null=True,
-        blank=True,
+        blank=True
+    )
+
+    trained_competence = models.CharField(
+        help_text=_(
+            'Trained competence e.g. Plugin development.'),
+            max_length=200,
```
Comment (ismailsunni) [inline django_project/certification/models/course.py]: __[Flake8]__

[E131] continuation line unaligned for hanging indent

Comment (sonlinux) [reply django_project/certification/models/course.py]: Thanks 
Updating.

**Comment by sonlinux** [comment]
Do we need to show the trained competence on the PDF certificate as well?

**Comment by ismailsunni** [comment]
Hi @sonlinux 
Just a suggestion, you can also rebase your commits so that it has a better/readable/more useful history. I usually use `git rebase -i HEAD^n` with `n` is the number of last commits.
For this PR, you can use it to merge the `clean code` commits.

**Comment by ann26** [comment]
hi @sonlinux , could you give a full screenie for the pdf certificate please? In your above screenie the signatures and footnote are gone.

**Comment by sonlinux** [comment]
Alright @ann26 
I will do so.

**Comment by ann26** [comment]
thanks @sonlinux ,
Please also resolve the conflicts and add unittest. The coverage is decreasing.

**Comment by sonlinux** [comment]
Thanks @ann26 
Ill do that.

**Comment by ismailsunni** [comment]
Flake8 detected 40 issues on 139da65e0fa373890691344acde1a5ab5f52f27b
Visit https://sideci.com/gh/10589087/pull_requests/867 to review the issues.

File: django_project/certification/views/certificate.py [hunk 4]
```diff
@@ -314,13 +315,19 @@ def generate_pdf(
             attendee.surname.encode('utf-8')))
     page.setFont('Times-Roman', 16)
     page.drawCentredString(
-        center, 360, 'Has attended and completed the course:')
+        center, 370, 'Has attended and completed the course:')
     page.setFont('Times-Bold', 20)
+    page.drawCentredString(center, 350, course.course_type.name.encode('utf-8))
```
Comment (ismailsunni) [inline django_project/certification/views/certificate.py]: __[Flake8]__

[E226] missing whitespace around arithmetic operator

Comment (ismailsunni) [inline django_project/certification/views/certificate.py]: __[Flake8]__

[E999] SyntaxError: EOL while scanning string literal

File: django_project/certification/tests/views/test_certificate_pdf.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+class TestGenerateCertificate(TestCase):
+    """Tests generation of a PDF certificate with expected content."""
+
+    @override_settings(VALID_DOMAIN=['testerver', ])
+    def setUp(self):
+        """Set up before test is run."""
+
+        self.client = Client()
+        self.client.post(
+				'/set_language/', data={'language': 'en'})
```
Comment (ismailsunni) [inline django_project/certification/tests/views/test_certificate_pdf.py]: __[Flake8]__

[E101] indentation contains mixed spaces and tabs

File: django_project/certification/tests/views/test_certificate_pdf.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        self.user = UserF.create(**{
```
Comment (ismailsunni) [inline django_project/certification/tests/views/test_certificate_pdf.py]: __[Flake8]__

[E101] indentation contains mixed spaces and tabs

**Pull request merged**